### ISSUE TITLE
Components with period in their names are removed unconditionally with the unusedComponents filter option.

### ISSUE THREAD

**Issue by lapinek** [thim81/openapi-format#37]
If a key under the fixed fields in the `components` object (in a descriptor) has a period, the key is removed from the descriptor after filtering with the `unusedComponents` option. For example:

`filter.json`
```json
{

    "unusedComponents": [
        "examples"
    ]
}
```

`descriptor.yaml`
```yaml
# . . .

      responses:
        '200':
          content:
            application/json:
              examples:
                tokenResponse:
                  $ref: '#/components/examples/Id.Token'
# . . .
components:
  examples:
    Id.Token:
# . . .
```

After filtering, the reference will remain, but the key in the `components` object will be removed.

If I read the spec correctly, periods are allowed in the key names; from https://spec.openapis.org/oas/v3.0.3#components-object:
> All the fixed fields declared above are objects that MUST use keys that match the regular expression: ^[a-zA-Z0-9\.\-_]+$.

```javascript
var regex = /^[a-zA-Z0-9\.\-_]+$/;
var key = 'Id.Token';
key.match(regex); // > [ "Id.Token" ]
```

**Comment by thim81** [comment]
@lapinek Thanks for reporting the issue and providing a clear example.

I'll try to look into it and try to reproduce it, so I can fix it.

**Comment by thim81** [comment]
@lapinek 
Short update, I was able to reproduce and I think that I know what is causing it (the component usage tracking logic).
I hope to find some time to implement a fix and release it next week.

**Comment by thim81** [comment]
@lapinek This bug should be resolved in release 1.9.0 of openapi-format.

Thanks for reporting it. I'm going to close the issue, feel free to open a new issue if you run into problems.
